refactor: fixing log related naming

benesjan opened the pull request:
Fixes #15012

The naming became strange with the introduction of the concept of a message that can be emitted either as a private log or as an offchain message - via an offchain effect (also used to be called out-of-band). Before we just used to have a private log.

### Changes
- Introduces "alias" for  `PRIVATE_LOG_CIPHERTEXT_LEN` called `MESSAGE_CIPHERTEXT_LEN` in `Aztec.nr`
- Renames `PRIVATE_LOG_PLAINTEXT_SIZE_IN_{BYTES,FIELDS}` → `MESSAGE_PLAINTEXT_SIZE_IN_{BYTES,FIELDS}`

benesjan commented:
* **#16882** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16882?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16882?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on barretenberg/cpp/src/barretenberg/vm2/common/protocol_contract_data.cpp:
```diff
@@ -5,7 +5,7 @@ namespace bb::avm2 {
 const std::unordered_map<CanonicalAddress, DerivedAddress> derived_addresses = {
     { CANONICAL_AUTH_REGISTRY_ADDRESS, AztecAddress("0x1c2474a77af3c756895867fa7622e5e9295121b35849e965f4614d51f00c62e2") },
-	{ CONTRACT_INSTANCE_REGISTRY_CONTRACT_ADDRESS, AztecAddress("0x001e627e78063a4944f1331b4c59544f8ad6d6d478f2fc79b025e8240b94b816") },
+	{ CONTRACT_INSTANCE_REGISTRY_CONTRACT_ADDRESS, AztecAddress("0x1f4284dea3e475d9d40283c7fe4c1aef5cdea59c012a8fa84c0c9ebe0752327e") },
```
For context see https://aztecprotocol.slack.com/archives/C02M7VC7TN0/p1757341750525609

nventuro commented on noir-projects/noir-protocol-circuits/crates/types/src/constants.nr:
```diff
@@ -271,10 +271,12 @@ pub global PRIVATE_LOG_LENGTH: u32 = PRIVATE_LOG_SIZE_IN_FIELDS + 1 /* length */
-/// because, unlike public logs, it is expected that private logs are encrypted (though this is not
-/// mandatory).
-pub global PRIVATE_LOG_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_SIZE_IN_FIELDS - 1; // -1 for the tag
+/// Messages are either delivered via a private log or emitted as an offchain message. The ciphertext length is derived
+/// from private log related constants because we want the message length to be consistent no matter the delivery mode
+/// (this allows us to reuse message processing functionality, see `do_process_message` function).
+///
+/// A private log contains the message ciphertext and the tag. Hence we subtract 1 for the tag.
+pub global MESSAGE_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_SIZE_IN_FIELDS - 1;
```
Not a fan of having message length in protocol circuits. The private log ciphertext is fine because the tag _is_ a protocol concern (since nodes index by tag etc), but _our_ usage of logs for messages is an aztec-nr thing. So I'd keep `PRIVATE_LOG_CIPHERTEXT_LEN` in here, and have definitions for `MESSAGE_CIPHER_LEN` in our message mod.

benesjan replied:
So that's how I quickly get a review from Nico - threatening with merging something he doesn't like 😂

Good points. Will rework this

nventuro commented on noir-projects/aztec-nr/aztec/src/messages/processing/log_retrieval_response.nr:
```diff
@@ -1,10 +1,9 @@
 
-global MAX_LOG_CONTENT_LEN: u32 =
-    std::cmp::max(PUBLIC_LOG_PLAINTEXT_LEN, PRIVATE_LOG_CIPHERTEXT_LEN);
+global MAX_LOG_CONTENT_LEN: u32 = std::cmp::max(PUBLIC_LOG_PLAINTEXT_LEN, MESSAGE_CIPHERTEXT_LEN);
```
See how this reads a bit crazy - why are we measuring public log plaintext with 'message ciphertext'? We should be comparing pub and priv logs, no these. Also, is the content encrypted or not? Because if not it shold be the private _plaintext_.

benesjan replied:
Completely reworked this and this change got reverted.

benesjan commented on noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr:
```diff
@@ -161,7 +160,7 @@ impl LogEncryption for AES128 {
     fn encrypt_log<let PlaintextLen: u32>(
```
This will be renamed in a followup PR to `MessageEncryption::encrypt`.

benesjan commented on noir-projects/aztec-nr/aztec/src/messages/encoding.nr:
```diff
@@ -1,9 +1,30 @@
+// utility function function that is inject by the #[aztec] macro.
+pub global MESSAGE_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_CIPHERTEXT_LEN;
+
+// TODO(#12750): The global variables below should not be here as they are AES128 specific.
+// ciphertext_length (2) + 14 bytes pkcs#7 AES padding.
+pub(crate) global HEADER_CIPHERTEXT_SIZE_IN_BYTES: u32 = 16;
+
+pub global EPH_PK_X_SIZE_IN_FIELDS: u32 = 1;
+pub global EPH_PK_SIGN_BYTE_SIZE_IN_BYTES: u32 = 1;
+
+// (17 - 1) * 31 - 16 - 1 = 479
+global MESSAGE_PLAINTEXT_SIZE_IN_BYTES: u32 = (MESSAGE_CIPHERTEXT_LEN - EPH_PK_X_SIZE_IN_FIELDS)
+    * 31
+    - HEADER_CIPHERTEXT_SIZE_IN_BYTES
+    - EPH_PK_SIGN_BYTE_SIZE_IN_BYTES;
+// Each field of the original note log was serialized to 32 bytes. Below we convert the bytes back to fields.
+// 479 / 32 = 15
+pub global MESSAGE_PLAINTEXT_SIZE_IN_FIELDS: u32 = MESSAGE_PLAINTEXT_SIZE_IN_BYTES / 32;
```
These values were moved in here from `log_encryption.nr` and renamed to reflect the message naming. This allowed me to hide the `MESSAGE_PLAINTEXT_SIZE_IN_BYTES` which I think is a good sign (meaning it was only used in this file and previously imported from `log_encryption.nr`).

Think this file being called `encoding.nr` makes this quite a natural place for these constants but if the reviewer have a better idea where to place these I am fine with reworking this.

The `HEADER_CIPHERTEXT_SIZE_IN_BYTES` being here is ugly but having that in `log_encryption.nr` was ugly as well so I think I didn't worsen it.

nventuro commented on noir-projects/aztec-nr/aztec/src/messages/encryption/log_encryption.nr:
```diff
@@ -25,12 +8,13 @@ pub global PRIVATE_LOG_PLAINTEXT_SIZE_IN_FIELDS: u32 = PRIVATE_LOG_PLAINTEXT_SIZ
-/// - `PRIVATE_LOG_CIPHERTEXT_LEN`: Fixed length of encrypted logs (defined globally)
-/// - `PRIVATE_LOG_PLAINTEXT_SIZE_IN_FIELDS`: Maximum size of decrypted plaintext (defined globally)
+/// - `MESSAGE_CIPHERTEXT_LEN`: Fixed length of encrypted message (defined globally)
+/// - `MESSAGE_PLAINTEXT_SIZE_IN_FIELDS`: Maximum size of decrypted plaintext (defined globally)
```
```suggestion
/// - `MAX_MESSAGE_PLAINTEXT_LEN`: Maximum size of decrypted plaintext (defined globally)
```

This creates a nice simmetry with `MESSAGE_PLAINTEXT_LEN` no? Without looking at this explanation it's not obvious to me what the difference between `MESSAGE_CIPHERTEXT_LEN` and `MESSAGE_PLAINTEXT_SIZE_IN_FIELDS` is - I'd guess that it has to do with fields vs bytes, but that's not it.

benesjan replied:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/16882/commits/e1f6ab97368d7eadf9907911e0f0407b8bbcba66

There is a bit of an inconsistency in naming. For length denominated in fields we use "_LEN" and for length denominated in bytes we use "_SIZE_IN_BYTES". The change would explode the scope of the PR so decided to not do it here and plan on just learning to live with it.

nventuro reviewed (approved):
Very nice. Thank you.

benesjan committed:
refactor: fixing log related naming

Fixes #15012

The naming became strange with the introduction of the concept of a message that can be emitted either as a private log or as an offchain message - via an offchain effect (also used to be called out-of-band). Before we just used to have a private log.

### Changes
- Introduces "alias" for  `PRIVATE_LOG_CIPHERTEXT_LEN` called `MESSAGE_CIPHERTEXT_LEN` in `Aztec.nr`
- Renames `PRIVATE_LOG_PLAINTEXT_SIZE_IN_{BYTES,FIELDS}` → `MESSAGE_PLAINTEXT_SIZE_IN_{BYTES,FIELDS}`
